linera-core/src/worker.rs

```diff
@@ -1391,106 +833,15 @@ where
-        }
-        if let Some(owner) = query.request_owner_balance {
-            info.requested_owner_balance =
-                chain.execution_state.system.balances.get(&owner).await?;
-        }
-        if let Some(next_block_height) = query.test_next_block_height {
-            ensure!(
-                chain.tip_state.get().next_block_height == next_block_height,
-                WorkerError::UnexpectedBlockHeight {
-                    expected_block_height: next_block_height,
-                    found_block_height: chain.tip_state.get().next_block_height
-                }
-            );
-        }
-        if query.request_pending_messages {
-            let mut messages = Vec::new();
-            let origins = chain.inboxes.indices().await?;
-            let inboxes = chain.inboxes.try_load_entries(&origins).await?;
-            let action = if *chain.execution_state.system.closed.get() {
-                MessageAction::Reject
-            } else {
-                MessageAction::Accept
-            };
-            for (origin, inbox) in origins.into_iter().zip(inboxes) {
-                for event in inbox.added_events.elements().await? {
-                    messages.push(IncomingMessage {
-                        origin: origin.clone(),
-                        event: event.clone(),
-                        action,
-                    });
-                }
-            }
-
-            info.requested_pending_messages = messages;
-        }
-        if let Some(range) = query.request_sent_certificates_in_range {
-            let start: usize = range.start.try_into()?;
-            let end = match range.limit {
-                None => chain.confirmed_log.count(),
-                Some(limit) => start
-                    .checked_add(usize::try_from(limit).map_err(|_| ArithmeticError::Overflow)?)
-                    .ok_or(ArithmeticError::Overflow)?
-                    .min(chain.confirmed_log.count()),
-            };
-            let keys = chain.confirmed_log.read(start..end).await?;
-            let certs = self.storage.read_certificates(keys).await?;
-            info.requested_sent_certificates = certs;
-        }
-        if let Some(start) = query.request_received_log_excluding_first_nth {
-            let start = usize::try_from(start).map_err(|_| ArithmeticError::Overflow)?;
-            info.requested_received_log = chain.received_log.read(start..).await?;
-        }
-        if let Some(hash) = query.request_hashed_certificate_value {
-            info.requested_hashed_certificate_value =
-                Some(self.storage.read_hashed_certificate_value(hash).await?);
-        }
-        if let Some(blob_id) = query.request_blob {
-            info.requested_blob = Some(self.storage.read_hashed_blob(blob_id).await?);
-        }
-        if query.request_manager_values {
-            info.manager.add_values(chain.manager.get());
+        let result = async move {
```

Comment:
why `async move {}.await;` ?

Reply:
Just removing the `async {}.await` block would make it not print the closing `trace` message. The alternative is to `.map` the error, which reads a bit differently. I'm not sure which one is best :thinking: